tls/s2n_connection.c

```diff
@@ -891,6 +891,7 @@ int s2n_connection_set_read_fd(struct s2n_connection *conn, int rfd)
     struct s2n_blob ctx_mem = {0};
     struct s2n_socket_read_io_context *peer_socket_ctx;
 
+    POSIX_ENSURE_REF(conn);
```

Comment:
Why are the new null checks necessary? I don't see us dereferencing `conn` here.

Reply:
I think 
conn->managed_io = 1;
can result in seg fault...

Reply:
done in the last commit